### ISSUE TITLE
Custom openapi.yml under META-INF not merged after upgrade to 3.17.x

### ISSUE THREAD

**Issue by luca-bassoricci** [quarkusio/quarkus#44886]
### Describe the bug

After upgrading to Quarkus 3.17 partial Openapi files (openapi.yml) under META-INF is not merged into final openapi.yml schema.
In version 3.16.4 file is merged es expected.
Partial openapi.yml
```
components:
  schemas:
    ListOfValues:
      type: string
      enum:
        - Area_condivisa
        - AC
        - ADCAM
        - CNM
        - IVATR
        - IRAP
        - MODIVA
        - UNIPF
        - UNISC
        - UNISB
        - UNISP
        - STSET
```
[openapi - not merged.txt](https://github.com/user-attachments/files/17989606/openapi.-.not.merged.txt)
[openapi - merged.txt](https://github.com/user-attachments/files/17989607/openapi.-.merged.txt)

### Expected behavior

---
openapi: 3.0.3
info:
  title: openapi-issue API
  version: 1.0.0-SNAPSHOT
paths:
  /openapi-issue:
    post:
      tags:
      - Resource
      description: Do Action
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/DoActionRequest"
      responses:
        "201":
          description: Created
components:
  schemas:
    ListOfValues:
      enum:
      - Area_condivisa
      - AC
      - ADCAM
      - CNM
      - IVATR
      - IRAP
      - MODIVA
      - UNIPF
      - UNISC
      - UNISB
      - UNISP
      - STSET
      type: string
    DoActionRequest:
      type: object
      properties:
        value:
          $ref: "#/components/schemas/ListOfValues"

### Actual behavior

---
components:
  schemas:
    DoActionRequest:
      type: object
      properties:
        value:
          $ref: "#/components/schemas/ListOfValues"
openapi: 3.1.0
paths:
  /openapi-issue:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/DoActionRequest"
        required: true
      responses:
        "201":
          description: Created
      summary: Do Action
      tags:
      - Resource
info:
  title: openapi-issue API
  version: 1.0.0-SNAPSHOT

### How to Reproduce?

Start with quarkus:dev and open swagger-ui console

### Output of `uname -a` or `ver`

Microsoft Windows [Versione 10.0.22631.4460]

### Output of `java -version`

java version "17.0.4.1" 2022-08-18 LTS

### Quarkus version or git rev

3.17.2

### Additional information

Reproducer here: https://github.com/luca-bassoricci/openapi-issue/
main branch for 3.17.2 behaviour, switch to branch 3.16.x for old one.

**Comment by MikeEdgar** [comment]
I'm looking into this.

**Comment by phillip-kruger** [comment]
@MikeEdgar maybe the versions ? The one in 3.0.3 and the other 4 ?

**Comment by MikeEdgar** [comment]
It's fixed in smallrye-open-api 4.0.4 - I'm releasing it now 😃 . The related PR was https://github.com/smallrye/smallrye-open-api/pull/2079, this is happening because the classes of the schema maps were not the same so the static file was dropped.

**Comment by gsmet** [comment]
This will be included in 3.17.3 released tomorrow (if all goes well :)).

Thanks a lot for the detailed report and the reproducer. It makes a huge difference!
